### PR TITLE
Removed statistics from the disaggregation calculator

### PR DESCRIPTION
**Pull request by micheles**
Instead of computing all the the realizations (and then the statistics) only one realization is computed. The user can specify it by setting `rlzi = XXX` in the job.ini. If left unspecified, the engine will use the closest realization to the mean hazard curve (NB: if there is more than 1 site the chosen realization will be site-dependent).
Some tests for Seattle (with 80 realizations) have shown a speedup of 42x and a reduction in the memory occupation of 15x. Correctness tests are required.

Incidentally closes https://github.com/gem/oq-engine/issues/4499

### CODE REVIEW

File: openquake/calculators/disaggregation.py [hunk 1]
```diff
@@ -355,47 +361,26 @@ def save_bin_edges(self):
...
         for sid in self.sitecol.sids:
             self.datastore['disagg-bins/lons/sid-%d' % sid] = b[2][sid]
             self.datastore['disagg-bins/lats/sid-%d' % sid] = b[3][sid]
         self.datastore['disagg-bins/eps'] = b[4]
 
-    def build_stats(self, results, hstats):
-        """
-        :param results: dict key -> 6D disagg_matrix
-        :param hstats: (statname, statfunc) pairs
+    def get_NPM(self):
```
Comment (mmpagani) [inline openquake/calculators/disaggregation.py]: Please add also here one line describing this method

File: openquake/calculators/disaggregation.py [hunk 2]
```diff
@@ -47,30 +46,34 @@ def _to_matrix(matrices, num_trts):
     return mat
 
 
-def gen_iml3s(R, iml_disagg, imtls, poes_disagg, curves):
+def _iml2s(rlzis, iml_disagg, imtls, poes_disagg, curves):
```
[the remark this answers is not in the document]
Comment (micheles) [reply openquake/calculators/disaggregation.py]: It is a private method, so I added a comment to it

**Comment by rcgee** [comment]
Using the closest realization to the mean hazard curve is very useful to speed up the disaggregation! Specifying a specific realization would probably not be used as much. 

Why not keep the original (full) disaggregation if  `rlzi =` is not given? And have the option of using the closest realization to the mean hazard curve when a specific flag is given, for example `rlzi = closest_mean` ? That way the two results can be compared.

What do you both think?

**Comment by micheles** [comment]
I am strongly against keeping the old behavior together with the new one, it would complicated the code a lot and go against further optimizations that I want to do in the future. It is always possible to to manually perform R calculations, one per realization and then manually compute the stats. It is inconvenient, but I see that as a plus. If the compare is to compare with the old approach during the transition phase, you can always switch the branch and go back to an older version of the engine.

**Comment by rcgee** [comment]
OK @micheles  LGTM

**Pull request merged**